src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Helpers.cs

```diff
@@ -418,12 +418,17 @@ public static byte[] GetSubjectKeyIdentifier(this X509Certificate2 certificate)
             //
             // https://msdn.microsoft.com/en-us/library/windows/desktop/aa376079%28v=vs.85%29.aspx
 
+            PublicKey key = certificate.PublicKey;
+            SubjectPublicKeyInfoAsn spki = new SubjectPublicKeyInfoAsn();
+            spki.Algorithm = new AlgorithmIdentifierAsn { Algorithm = key.Oid, Parameters = key.EncodedParameters.RawData };
+            spki.SubjectPublicKey = key.EncodedKeyValue.RawData;
+
```

Comment:
I can actually simplify it even further by using `new X509SubjectKeyIdentifierExtension(certificate.PublicKey, X509SubjectKeyIdentifierHashAlgorithm.CapiSha1, false).RawData`. Would that be preferred?